pkg/volume/downwardapi/downwardapi_test.go

```diff
@@ -191,6 +196,74 @@ func TestDownwardAPI(t *testing.T) {
 	}
 }
 
+// TestDownwardAPICollectDataFsUser tests as much as possible, without running
+// with privileges, that CollectData() does what is expected regarding FsUser.
+// We test FsUser like this, because extending TestDownwardAPI() to create the
+// files with a different user needs privileges (to chown it to a differet user)
+// and we run without privileges in the CI.
+func TestDownwardAPICollectDataFsUser(t *testing.T) {
+	fsUser := int64(100)
+	mode := int32(0644)
+	cases := []struct {
+		name           string
+		featureEnabled bool
+		FsUser         *int64
+		success        bool
+		item           v1.DownwardAPIVolumeFile
+		expProj        map[string]volumeutil.FileProjection
+	}{
+		{
+			name:           "basic UserNamespacesSupport enabled",
+			featureEnabled: true,
+			FsUser:         &fsUser,
+			success:        true,
+			item: v1.DownwardAPIVolumeFile{
+				Path: "test",
+			},
+			expProj: map[string]volumeutil.FileProjection{
+				"test": {
+					FsUser: &fsUser,
+					Mode:   mode,
+				},
+			},
+		},
+		{
+			name:           "basic UserNamespacesSupport disabled",
+			featureEnabled: false,
+			FsUser:         &fsUser,
+			success:        true,
+			item: v1.DownwardAPIVolumeFile{
+				Path: "test",
+			},
+			expProj: map[string]volumeutil.FileProjection{
+				"test": {
+					Mode: mode,
+				},
+			},
+		},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.UserNamespacesSupport, tc.featureEnabled)()
+			vol := volumetest.NewFakeVolumeHost(t, "/tmp/non-existent-dir/", nil, nil)
+			items := []v1.DownwardAPIVolumeFile{tc.item}
+
+			proj, err := CollectData(items, nil, vol, &mode, tc.FsUser)
+
+			if tc.success && err != nil {
+				t.Errorf("Expected success, but got error: %v", err)
+			}
+			if !tc.success && err == nil {
+				t.Error("Expected error, but got success")
+			}
+			if !reflect.DeepEqual(proj, tc.expProj) {
```

Comment:
nit: (not worth resubmitting for by itself) consider `require.Equal(t, tc.expProj, proj)` - it will highlight the diff within the structures